.github/workflows/conda-cpp-build.yaml

```diff
@@ -39,23 +42,40 @@ permissions:
   statuses: none
 
 jobs:
+  compute-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      MATRIX: ${{ steps.compute-matrix.outputs.MATRIX }}
+    steps:
+      - name: Compute Build Matrix
+        id: compute-matrix
+        run: |
+          set -eo pipefail
+
+          export MATRIX='
+          - { CUDA_VER: "11.8.0", LINUX_VER: "ubuntu22.04", ARCH: "amd64", PY_VER: "3.10" }
+          - { CUDA_VER: "11.8.0", LINUX_VER: "ubuntu22.04", ARCH: "arm64", PY_VER: "3.10" }
+          '
+
+          echo "MATRIX=$(
+            yq -n -o json 'env(MATRIX)' | \
+            jq -c '${{ inputs.matrix_filter }} | {includes: .}' \
+          )" >> ${GITHUB_OUTPUT}
   build:
+    needs: compute-matrix
     timeout-minutes: 480
     strategy:
       fail-fast: false
-      matrix:
-        include:
-          - { CUDA_VER: "11.8.0", LINUX_VER: "ubuntu22.04", ARCH: "amd64", PY_VER: "3.10" }
-          - { CUDA_VER: "11.8.0", LINUX_VER: "ubuntu22.04", ARCH: "arm64", PY_VER: "3.10" }
+      matrix: ${{ fromJSON(needs.compute-matrix.outputs.MATRIX) }}
     runs-on:
       - self-hosted
       - linux
       - ${{ inputs.node_type }}
-      - ${{ matrix.ARCH }}
+      - ${{ matrix.includes.ARCH }}
     env:
       RAPIDS_ARTIFACTS_DIR: ${{ github.workspace }}/artifacts
     container:
-      image: rapidsai/ci:cuda${{ matrix.CUDA_VER }}-${{ matrix.LINUX_VER }}-py${{ matrix.PY_VER }}
+      image: rapidsai/ci:cuda${{ matrix.includes.CUDA_VER }}-${{ matrix.includes.LINUX_VER }}-py${{ matrix.includes.PY_VER }}
```

Comment:
Is it possible to remove the "includes" level of key here? I am not super familiar with `jq` syntax -- it looks like this is a level of hierarchy that is being added but isn't strictly necessary?

Reply:
Yes, you're right. I battled with this in the past and couldn't figure out why it was necessary for dynamic matrices but not hardcoded matrices.

After re-reading the GHAs docs below, I'm now realizing that my matrix output should say `include` and not `includes`. I think once I fix that typo in the first job, we can remove this extra level in the subsequent job.

- https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations

I'll push a follow-up commit to test.

Reply:
Yup, that worked in the downstream https://github.com/rapidsai/ucxx/pull/18 PR.

Good catch @bdice!